src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +743,26 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	@Override
+	public void removeNodeFromParent(MutableTreeNode node) {
+		super.removeNodeFromParent(node);
+		if(node.getParent() == getRoot()) {
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), 
+	        		new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_REMOVED_EVENT, new Target((SiteNode)node)));
+		} else {
+			ZAP.getEventBus().publishSyncEvent(SiteMapEventPublisher.getPublisher(), 
+	        		new Event(SiteMapEventPublisher.getPublisher(), SiteMapEventPublisher.SITE_NODE_REMOVED_EVENT, new Target((SiteNode)node)));
+		}
+	}
 
+	private void publishAddEvent(SiteNode parent, SiteNode item) {

```

Comment:
btw, some statements/lines (in both methods) are indented with spaces and others with tabs, this should be normalised.